libs/clients/work-machines/src/lib/providers.ts

```diff
@@ -1,14 +1,93 @@
+import {
+  XRoadConfig,
+  IdsClientConfig,
+  LazyDuringDevScope,
+} from '@island.is/nest/config'
+import { ConfigType } from '@nestjs/config'
 import {
   Configuration,
+  MachineCategoryApi,
+  MachineOwnerChangeApi,
+  MachineSupervisorChangeApi,
   MachinesApi,
   MachinesDocumentApi,
 } from '../../gen/fetch'
-import { ApiConfig } from './api.config'
+import { WorkMachinesClientConfig } from './workMachines.config'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+const ConfigFactory = (
+  xroadConfig: ConfigType<typeof XRoadConfig>,
+  config: ConfigType<typeof WorkMachinesClientConfig>,
+  idsClientConfig: ConfigType<typeof IdsClientConfig>,
+  acceptHeader: string,
+) => ({
+  fetchApi: createEnhancedFetch({
+    name: 'clients-work-machines',
+    organizationSlug: 'vinnueftirlitid',
+    logErrorResponseBody: true,
+    autoAuth: idsClientConfig.isConfigured
+      ? {
+          mode: 'tokenExchange',
+          issuer: idsClientConfig.issuer,
+          clientId: idsClientConfig.clientId,
+          clientSecret: idsClientConfig.clientSecret,
+          scope: config.fetch.scope,
+        }
+      : undefined,
+  }),
+  basePath: `${xroadConfig.xRoadBasePath}/r1/${config.xRoadServicePath}`,
+  headers: {
+    'X-Road-Client': xroadConfig.xRoadClient,
+    Accept: acceptHeader,
+  },
+})
 
-export const apiProviders = [MachinesApi, MachinesDocumentApi].map((Api) => ({
-  provide: Api,
-  useFactory: (configuration: Configuration) => {
-    return new Api(configuration)
+export class CustomMachineApi extends MachinesApi {}
+
+export const apiProviders = [
+  {
+    api: MachinesApi,
+    provide: MachinesApi,
+    acceptHeader: 'application/vnd.ver.machines.hateoas.v1+json',
+  },
+  {
+    api: MachinesApi,
+    provide: CustomMachineApi,
+    acceptHeader: 'application/vnd.ver.machine.hateoas.v1+json',
+  },
+  {
+    api: MachineOwnerChangeApi,
+    provide: MachineOwnerChangeApi,
+    acceptHeader: 'application/json',
+  },
+  {
+    api: MachineSupervisorChangeApi,
+    provide: MachineSupervisorChangeApi,
+    acceptHeader: 'application/json-patch+json',
+  },
+  {
+    api: MachineCategoryApi,
+    provide: MachineCategoryApi,
+    acceptHeader: 'application/json',
+  },
+  {
+    api: MachinesDocumentApi,
+    provide: MachinesDocumentApi,
+    acceptHeader: 'application/vnd.ver.hateoas.v1+json',
+  },
+].map((item) => ({
```

Comment:
~ FYI you could use object destructuring to access the properties directly and skip using `item.` and use the property names directly.
```suggestion
].map(({api, provide, acceptHeader}) => ({
```

Reply:
Applied